encrypt/encrypt.go

```diff
@@ -75,15 +82,37 @@ func Encrypt(args []string) error {
 		return err
 	}
 
-	// no key provided, check for one in the session file
+	var sesKey string
 	if len(publicKeyFileList) == 0 {
-
-		sesKey, err := helpers.GetPublicKey()
+		// check for public key in .sda-cli-session file from login
+		sesKey, err = helpers.GetPublicKeyFromSession()
 		if err != nil {
-			return fmt.Errorf("public key not provided or %v", err)
+			log.Println("could not read key from previous login,", err)
 		}
+	}
+	// key from session file found
+	if len(publicKeyFileList) == 0 && sesKey != "" {
 		publicKeyFileList = append(publicKeyFileList, sesKey)
 	}
+	if len(publicKeyFileList) == 0 && sesKey == "" && *target != "" {
+		// fetch info endpoint values
+		info, err := login.GetAuthInfo(*target)
+		if err != nil {
+			return err
+		}
+		// create pub file
+		pubFile, err := helpers.CreatePubFile(info.PublicKey, "crypt4gh_key.pub")
+		if err != nil {
+			return err
+		}
+		log.Println("fetching public key")
+		// no key provided, no key in session file, target provided
+		publicKeyFileList = append(publicKeyFileList, pubFile)
+	}
+	// no key provided, no key in session file, no target provided
+	if len(publicKeyFileList) == 0 && sesKey == "" && *target == "" {
+		return errors.New("no public key could be obtained")
+	}
```

Comment:
This entire block makes no logical sense.
 `GetPublicKeyFromSession` will return the name of the public key or an error.
That means that this block can be simplified to two if blocks:
```
if publicKeyFileList == nil && *target == "" {
...
```
and
```
if publicKeyFileList == nil && *target != "" {
...
```
or as a switch statement:
```
if publicKeyFileList == nil {
	switch len(*target) {
	case 0:
...
```